common/persistence/history_manager.go

```diff
@@ -171,6 +171,7 @@ func (m *executionManagerImpl) DeleteHistoryBranch(
 			// skip the target branch
 			continue
 		}
+		usedBranches[br.BranchId] = common.LastEventID
```

Comment:
won't this prevent anything from been deleted in the tree except the target branch?

Reply:
If target branch's ancestor is not reference by other branches, those ancestors will also be deleted. Not sure if this is your question.

Reply:
Synced offline.